dataplane/src/nat/stateful/mod.rs

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+#![allow(unused_variables)]
+
+mod allocator;
+mod sessions;
+
+use super::Nat;
+use net::headers::Net;
+use net::vxlan::Vni;
+use std::net::IpAddr;
+
+#[derive(thiserror::Error, Debug)]
+pub enum StatefulNatError {
+    #[error("other error")]
+    Other,
+}
+
+#[derive(Debug, Clone)]
+struct NatState {}
+
+#[derive(Debug, Clone)]
+struct NatTuple {
+    src_ip: IpAddr,
+    dst_ip: IpAddr,
+    protocol: u8,
+    vni: Vni,
+}
+
+impl NatState {
+    fn new(net: &Net, pool: &allocator::NatPool) -> Self {
+        Self {}
+    }
+}
+
+impl Nat {
+    fn hash_tuple(net: &Net, vni: Vni) -> NatTuple {
```

Comment:
I'm not sure I understand the intent of this function.  Is it a lookup?  If so, what if it misses?  Don't we need to return `Option`

Reply:
> Is it a lookup? If so, what if it misses? Don't we need to return `Option`

Yes, that's right. Another detail in the skeleton that can be adjusted when the proper implementation lands (but it looks better if it's in order now, so I'll fix it)

Reply:
> I'm not sure I understand the intent of this function.

Looks like I moved this bit and forgot to rename it. This function is used to extract the tuple from the packet, so yes it's a lookup. `hash_` is not a good prefix here, I'll rename it, thank you for noticing :slightly_smiling_face: